src/mixins/maskable.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Maskable
+ *
+ * @mixin
+ *
+ * Creates an input mask that is
+ * generated from a masked str
+ *
+ * Example: mask="#### #### #### ####"
+ */
+export default {
+  data: () => ({
+    allowedMasks: ['#', 'A', 'a', 'X'],
+    selection: 0,
+    preDefined: {
+      'credit-card': '#### - #### - #### - ####',
+      'date': '##/##/####',
+      'date-with-time': '##/##/#### ##:##',
+      'phone': '(###) ### - ####',
+      'social': '###-##-####',
+      'time': '##:##',
+      'time-with-seconds': '##:##:##'
+    }
+  }),
+
+  props: {
+    fillMaskBlanks: Boolean,
+    mask: {
+      type: [Object, String],
+      default: null
+    },
+    returnMaskedValue: Boolean
+  },
+
+  computed: {
+    masked () {
+      const preDefined = this.preDefined[this.mask]
+      const mask = preDefined || this.mask || ''
+
+      return mask.split('')
+    }
+  },
+
+  methods: {
+    maskText (text, newText = []) {
```

Comment:
Last but not least - as discussed before I'd suggest to extract the mask logic to separate file (src/util/*), it will follow the single responsibility rule and make the code analysis or testing easier